src/main/java/com/visenze/datatracking/data/Event.java

```diff
@@ -0,0 +1,641 @@
+
+    public void setOs(String os) {
+        this.os = os;
+    }
+
+    public void setOsVersion(String osVersion) {
+        this.osVersion = osVersion;
+    }
+
+    public void setScreenResolution(String screenResolution) {
+        this.screenResolution = screenResolution;
+    }
+
+    public void setAppId(String appId) {
+        this.appId = appId;
+    }
+
+    public void setAppName(String appName) {
+        this.appName = appName;
+    }
+
+    public void setAppVersion(String appVersion) {
+        this.appVersion = appVersion;
+    }
+
+    public void setDeviceBrand(String deviceBrand) {
+        this.deviceBrand = deviceBrand;
+    }
+
+    public void setDeviceModel(String deviceModel) {
+        this.deviceModel = deviceModel;
+    }
+
+    public void setAaid(String aaid) {
+        this.aaid = aaid;
+    }
+
+    public void setDidmd5(String didmd5) {
+        this.didmd5 = didmd5;
+    }
+
+    public void setGeo(String geo) {
+        this.geo = geo;
+    }
+
+    public void setLang(String lang) {
+        this.lang = lang;
+    }
+
+    public void setCarrier(String carrier) {
+        this.carrier = carrier;
+    }
+
+    public void setCampaign(String campaign) {
+        this.campaign = campaign;
+    }
+
+    public void setCampaignAdGroup(String campaignAdGroup) {
+        this.campaignAdGroup = campaignAdGroup;
+    }
+
+    public void setCampaignCreative(String campaignCreative) {
+        this.campaignCreative = campaignCreative;
+    }
+
+    public void setS1(String s1) {
+        this.s1 = s1;
+    }
+
+    public void setS2(String s2) {
+        this.s2 = s2;
+    }
+
+    public void setS3(String s3) {
+        this.s3 = s3;
+    }
+
+    public void setS4(String s4) {
+        this.s4 = s4;
+    }
+
+    public void setS5(String s5) {
+        this.s5 = s5;
+    }
+
+    @SerializedName("cat")
+    @Expose
+    private String category;
+
+    @SerializedName("action")
+    @Expose
+    private String action;
+
+    @SerializedName("name")
+    @Expose
+    private String name;
+
+    @SerializedName("ts")
+    @Expose
+    private String timestamp;
+
+    @SerializedName("value")
+    @Expose
+    private String value;
+
+    @SerializedName("label")
+    @Expose
+    private String label;
+
+    @SerializedName("queryId")
+    @Expose
+    private String queryId;
+
+    @SerializedName("fromReqId")
+    @Expose
+    private String fromReqId;
+
+    @SerializedName("uid")
+    @Expose
+    private String uid;
+
+    @SerializedName("sid")
+    @Expose
+    private String sessionId;
+
+    @SerializedName("source")
+    @Expose
+    private String source;
+
+    @SerializedName("v")
+    @Expose
+    private String version;
+
+    @SerializedName("sdk")
+    @Expose
+    private String sdk;
+
+    @SerializedName("pid")
+    @Expose
+    private String pid;
+
+    @SerializedName("imUrl")
+    @Expose
+    private String imageUrl;
+
+    @SerializedName("pos")
+    @Expose
+    private String position;
```

Comment:
why position is String instead of "int" ?

Reply:
change to Integer if change to int, as int is primary type it will be serialized to default value 0.